### PR TITLE
304 ug delivery list delivery view

### LINKED ISSUE #304: UG - Delivery List, Delivery View

### PR DESCRIPTION
**Pull request by juliusgambe**
Closes #304

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -408,46 +409,86 @@ DELIVERY_DATE)!_
...
+* `delivery list --status SHIPPED --date TODAY`{.swift} </br>
+  Lists all deliveries with status `SHIPPED`{.swift} for all customers and expected delivery date of today in ascending
+  order.
 
-**Command failed (0 deliveries):** _There are currently no deliveries!_
+<box type="tip" background-color="#d9edf7" border-color="#bce8f1" icon=":bulb:">
 
-### View details of deliveries
+**Tip:**
+You combine any of the filters and sort options to get the list of deliveries that you want.
```
Comment (B-enguin) [inline docs/UserGuide.md]: Is this missing a verb, e.g. `You may combine` or `You can combine`

Comment (B-enguin) [reply docs/UserGuide.md]: Not sure if filter should be singular in this case also

File: docs/UserGuide.md [hunk 2]
```diff
@@ -408,46 +409,86 @@ DELIVERY_DATE)!_
...
+[&uarr; Back to Table of Contents](#top)
 
-_DELIVERY_ID_: Integer
+### View details of a delivery
 
-**Command succeeds:**
+> Shows the details of the specified delivery.
+
+If you would like to get the details of a specific delivery, you can use `delivery view`{.swift} to view the details of
+the delivery. You can see the delivery id, delivery name, delivery status, address, customer name customer ID, ordered
```
Comment (B-enguin) [inline docs/UserGuide.md]: Is there a missing comma between customer name and customer ID

File: docs/UserGuide.md [hunk 3]
```diff
@@ -408,46 +409,86 @@ DELIVERY_DATE)!_
...
+
+**Note:**
+
+* All fields are required.
+* The `DELIVERY_ID`{.swift} must be an integer.
+  </box>
+
+**Example:**
+
+* `delivery view 1001`{.swift} </br>
```
Comment (B-enguin) [inline docs/UserGuide.md]: Would it be more apt to show delivery view 1 instead inline with the success image

**Review by B-enguin — COMMENTED** [review]
Some general comments

File: docs/UserGuide.md [hunk 4]
```diff
@@ -408,46 +409,86 @@ DELIVERY_DATE)!_
...
+### View a list of deliveries
 
-Shows a list of all deliveries.
+> Lists the deliveries in the delivery book.
 
-**Format:** `delivery list [--status STATUS] [--customer CUSTOMER_ID] [--date EXPECTED_DELIVERY_DATE]  [--sort SORT]`
+If you would like to get an overview of all your deliveries, you can use `delivery list`{.swift} to list all your
+deliveries at once. You can also filter the list of deliveries by status, customer ID and expected delivery date.
+You can also sort the list of deliveries by expected delivery date in ascending or descending order. By default, the
+list of deliveries will be sorted by expected delivery date in descending order (latest first).
```
Comment (jianyangg) [inline docs/UserGuide.md]: Will this be italicised?

File: docs/UserGuide.md [hunk 5]
```diff
@@ -408,46 +409,86 @@ DELIVERY_DATE)!_
...
 
-_DELIVERY_ID_: Integer
+### View details of a delivery
 
-**Command succeeds:**
+> Shows the details of the specified delivery.
+
+If you would like to get the details of a specific delivery, you can use `delivery view`{.swift} to view the details of
+the delivery. You can see the delivery id, delivery name, delivery status, address, customer name customer ID, ordered
+date, expected delivery date, and notes (if any).
```
Comment (jianyangg) [inline docs/UserGuide.md]: Should these be italicised?

File: docs/UserGuide.md [hunk 6]
```diff
@@ -408,46 +409,86 @@ DELIVERY_DATE)!_
...
+**Before:**
+![](images/delivery/DeliveryListSuccess.png)
+
+**After:**
 
-![](images/delivery/delivery_view.png)
+![](images/delivery/DeliveryViewSuccess.png)
 
-**Command failed (0 deliveries):** _There are currently no deliveries._
+[&uarr; Back to Table of Contents](#top)
```
Comment (jianyangg) [inline docs/UserGuide.md]: Not sure if this `#top` link works. Should it be `#table-of-contents` instead?

File: docs/UserGuide.md [hunk 7]
```diff
@@ -408,46 +409,86 @@ DELIVERY_DATE)!_
...
 
-**Format:** `delivery view DELIVERY_ID`
+![](images/customer/customerList.png)
 
-**Example:** `delivery view 1001`
+**After:**
+![](images/delivery/DeliveryListSuccess.png)
 
-**Accepted Values:**
+[&uarr; Back to Table of Contents](#top)
```
Comment (jianyangg) [inline docs/UserGuide.md:742]: Is it `#table-of-contents` instead?

File: docs/UserGuide.md [hunk 8]
```diff
@@ -408,46 +409,86 @@ DELIVERY_DATE)!_
...
-_SORT_: String of either `asc` for ascending or `desc` for descending or defaults to sort by expected delivery date.
+* `delivery list --status CREATED --customer 1 --date 2023-12-12 --sort DESC`{.swift} </br>
+  Lists all deliveries with status `CREATED`{.swift} for customer with ID `1`{.swift} and expected delivery date of 12
+  Dec 2023 in descending order.
 
-**Command succeeds (>0 deliveries):**
-![](images/delivery/delivery_list.png)
+* `delivery list --status SHIPPED --date TODAY`{.swift} </br>
+  Lists all deliveries with status `SHIPPED`{.swift} for all customers and expected delivery date of today in ascending
+  order.
```
Comment (jianyangg) [inline docs/UserGuide.md]: If it's all of the same date, its sorted by ascending order of which characteristic? Is it delivery name?

**Review by jianyangg — COMMENTED** [review]
Few nits to fix

File: docs/UserGuide.md (hunk 4 above)
Comment (juliusgambe) [reply docs/UserGuide.md]: I think we shouldn't italicise this as we usually use italics to highlight or provide extra information / context.

File: docs/UserGuide.md (hunk 6 above)
Comment (juliusgambe) [reply docs/UserGuide.md]: Yes it does

File: docs/UserGuide.md (hunk 8 above)
Comment (juliusgambe) [reply docs/UserGuide.md]: Actually, it doesn't make sense to specify sorting here as it has been filtered by 1 date already.

**Review by jianyangg — APPROVED** [review]
LGTM!

**Pull request merged**